Dockerfile

```diff
@@ -1,37 +1,43 @@
+# Repo name: arynai/remote-processor-service
 
+##########
+# Common: resolve dependencies
+FROM python:3.11 AS rps_common
 
+ARG RPS_PORT=2796
 ARG POETRY_NO_INTERACTION=1
 ARG POETRY_VIRTUALENVS_IN_PROJECT=1
-ARG POETRY_VIRTUALENVS_CREATE=1 \
+ARG POETRY_VIRTUALENVS_CREATE=1
 ARG POETRY_CACHE_DIR=/tmp/poetry_cache
-ARG RPS_PORT=2796
-
-##########
-# Common: resolve dependencies
-FROM python:3.11 AS rps_common
 
 # Is there some way to keep this common layer common across all our services?
 # E.g. maybe we can have an image called 'aryn_service_base' or something
 # - setup aryn user and directory
 # - install commonly used software (poetry, maybe protoc)
 # And then we can just layer services on top?
-WORKDIR /aryn
+
+WORKDIR /aryn/
 COPY ./Makefile ./
 RUN make aryn_user
-RUN make install_poetry
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    make install_poetry
 
 USER aryn
 WORKDIR /aryn/rps/
 COPY --chown=aryn:aryn ./poetry.lock ./pyproject.toml ./
-RUN make -f ../Makefile common_build
+RUN --mount=type=cache,id=cache_poetry_1000,target=/tmp/poetry_cache,uid=1000,gid=1000,sharing=locked \
```

Comment:
/tmp/poetry_cache -> ${POETRY_CACHE_DIR}?

Reply:
I'd like it to be hard coded because I don't know what happens if you move the cache to a different directory.  It probably works, but it doesn't seem worth supporting.